### PR TITLE
Ret 2329

### PR DESCRIPTION
**Pull request by btensay**
Added PMD static code analyser in the Test project. The code changes are a mix of PMD suppressions and implementations of fixes for minor issues PMD flagged up. Refactoring and code improvement fixes for issues the suppressions cover will follow in subsequent tasks/tickets.

https://tools.hmcts.net/jira/browse/RET-2329

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingVenueHelperTest.java [hunk 1]
```diff
@@ -15,6 +15,9 @@
 import static uk.gov.hmcts.ecm.common.helpers.ESHelper.LISTING_VENUE_FIELD_NAME;
 import static uk.gov.hmcts.ecm.common.model.helper.Constants.ALL_VENUES;
 
+@SuppressWarnings({"PMD.UseProperClassLoader", "PMD.LawOfDemeter", "PMD.LinguisticNaming",
+    "PMD.ExcessiveMethodLength", "PMD.UnusedPrivateMethod", "PMD.JUnit5TestShouldBePackagePrivate",
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingVenueHelperTest.java:19]: JUnit5TestShouldBePackagePrivate - is this required?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelperTest.java [hunk 2]
```diff
@@ -68,7 +68,10 @@
 import static uk.gov.hmcts.ethos.replacement.docmosis.reports.Constants.NO_CHANGE_IN_CURRENT_POSITION_REPORT;
 import static uk.gov.hmcts.ethos.replacement.docmosis.reports.Constants.RESPONDENTS_REPORT;
 
-public class ListingHelperTest {
+@SuppressWarnings({"PMD.UseProperClassLoader", "PMD.LawOfDemeter", "PMD.AvoidInstantiatingObjectsInLoops",
+    "PMD.LinguisticNaming", "PMD.ExcessiveMethodLength", "PMD.NcssCount", "PMD.JUnit5TestShouldBePackagePrivate",
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelperTest.java:72]: This is simple to fix
PMD.JUnit5TestShouldBePackagePrivate
It shouldn't be suppressed

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/MultipleUtil.java [hunk 3]
```diff
@@ -67,6 +67,9 @@
 import static uk.gov.hmcts.et.common.model.multiples.MultipleConstants.HEADER_4;
 import static uk.gov.hmcts.et.common.model.multiples.MultipleConstants.SHEET_NAME;
 
+@SuppressWarnings({"PMD.LooseCoupling", "PMD.LawOfDemeter", "PMD.LinguisticNaming",
+    "PMD.ExcessiveMethodLength", "PMD.UseUnderscoresInNumericLiterals", "PMD.JUnit5TestShouldBePackagePrivate",
+    "PMD.ConfusingTernary", "PMD.CloseResource"})
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/MultipleUtil.java:72]: PMD.JUnit5TestShouldBePackagePrivate

Why is this even in here??? This isn't a test class!!!

Comment (btensay) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/MultipleUtil.java:72]: Yes it is not a test class. But it already exist in the test project and there are test classes that depend on it. Adjusting the suppressions and some of them are required.

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingVenueHelperTest.java (hunk 1 above)
Comment (btensay) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingVenueHelperTest.java:19]: Sorted

**Review by jthmcts — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Pull request merged**